sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -169,8 +169,9 @@ object GpuCast extends Arm {
   private val TIMESTAMP_REGEX_YYYY_MM = "\\A\\d{4}\\-\\d{1,2}[ ]?\\Z"
   private val TIMESTAMP_REGEX_YYYY = "\\A\\d{4}[ ]?\\Z"
   private val TIMESTAMP_REGEX_FULL =
-    "\\A\\d{4}\\-\\d{1,2}\\-\\d{1,2}[ T]?(\\d{1,2}:\\d{1,2}:\\d{1,2}\\.\\d{6}Z)\\Z"
-  private val TIMESTAMP_REGEX_NO_DATE = "\\A[T]?(\\d{1,2}:\\d{1,2}:\\d{1,2}\\.\\d{6}Z)\\Z"
+    "\\A\\d{4}\\-\\d{1,2}\\-\\d{1,2}[ T]?(\\d{1,2}:\\d{1,2}:([0-5]\\d|\\d)(\\.\\d{0,6})?Z?)\\Z"
```

Comment:
Consider transitioning the changed regex strings and new regex strings to a more readable form by using raw strings that are much more readable IMO.

```scala
scala> "\\A\\d{4}\\-\\d{1,2}\\-\\d{1,2}[ T]?(\\d{1,2}:\\d{1,2}:([0-5]\\d|\\d)(\\.\\d{0,6})?Z?)\\Z" == """\A\d{4}\-\d{1,2}\-\d{1,2}[ T]?(\d{1,2}:\d{1,2}:([0-5]\d|\d)(\.\d{0,6})?Z?)\Z"""
res1: Boolean = true
```

Reply:
The string is copied between this scala file and the testing site (https://regex101.com/). Such format is the result after copying from that site thus I am inclined to keep it to avoid modifying the entire string afterward.

Reply:
What if regex101 starts producing raw strings for Java 15 ? or goes out of business :) ?

Reply:
Interesting question :smile: